project/apps/api-gateway/src/auth/auth.guard.ts

```diff
@@ -0,0 +1,32 @@
+import {
+  Injectable,
+  CanActivate,
+  ExecutionContext,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { SupabaseService } from '../supabase/supabase.service';
+
+@Injectable()
+export class AuthGuard implements CanActivate {
```

Comment:
jus to check my understanding, this `AuthGuard` only checks if there's a valid session token right? doesn't differentiate between the users' roles.

Reply:
yup the roles i havent settle yet but we need an authguard before we handle roles. i have already created a roles column in the DB for us to populate. idea is for admin users we will manually set them in the DB.(but flexible to changes in future)

Reply:
yea, as for storing role in JWT i never knew u could do that but it seems like JWT main purpose should be for authentication rather than authorization. storing roles might lead to security issues - quoted from article https://stackoverflow.com/questions/47224931/is-setting-roles-in-jwt-a-best-practice